sdk/src/main/scala/com/horizen/validation/ConsensusValidator.scala

```diff
@@ -0,0 +1,82 @@
+package com.horizen.validation
+import java.time.Instant
+
+import com.horizen.SidechainHistory
+import com.horizen.block.SidechainBlock
+import com.horizen.chain.SidechainBlockInfo
+import com.horizen.consensus.{NonceConsensusEpochInfo, _}
+import scorex.core.block.Block
+import scorex.util.ScorexLogging
+
+import scala.util.Try
+
+class ConsensusValidator extends HistoryBlockValidator with ScorexLogging {
+  override def validate(block: SidechainBlock, history: SidechainHistory): Try[Unit] = Try {
+    if (history.isGenesisBlock(block.id)) {
+      validateGenesisBlock(block, history)
+    }
+    else {
+      validateNonGenesisBlock(block, history)
+    }
+  }
+
+  private def validateGenesisBlock(block: SidechainBlock, history: SidechainHistory): Unit = {
+    if (block.timestamp != history.params.sidechainGenesisBlockTimestamp) {
+      throw new IllegalArgumentException(s"Genesis block timestamp ${block.timestamp} is differ than expected timestamp from configuration ${history.params.sidechainGenesisBlockTimestamp}")
+    }
+
+    val vrfSignIsNotCorrect = false //shall be implemented as VRF key will be used for block signing
```

Comment:
we should call verifyVfr and verifyForgerBox with consensusEpochInfo calculated from SC creation TX and a constant for nonce